rust/psibase_macros/src/graphql_macro.rs

```diff
@@ -28,19 +27,44 @@ impl Parse for Args {
 
         let key_fn = input.parse().expect("Table key function not present");
 
-        let subindex_params = if input.parse::<Token![,]>().is_ok() {
-            Punctuated::<TypeParam, Token![,]>::parse_terminated(input)?
-                .into_iter()
-                .collect()
-        } else {
-            Vec::new()
-        };
+        let mut subindex_params = Vec::new();
+        let mut subindex_remaining_key_ty: Option<TypeParamBound> = None;
+
+        if input.parse::<Token![,]>().is_ok() {
+            loop {
+                if input.is_empty() {
+                    break;
+                } else if input.peek(Token![,]) {
+                    input.parse::<Token![,]>().expect("expected ,");
```

Comment:
The `expect` will always pass, because `peek` just checked the same thing.

Reply:
Good catch, will unwrap it.

Reply:
Actually with my new change it's irrelevant.